hpx/traits/is_callable_with.hpp

```diff
@@ -0,0 +1,38 @@
+//  Copyright (c) 2016 Lukas Troska
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_TRAITS_IS_CALLABLE_WITH_HPP
+#define HPX_TRAITS_IS_CALLABLE_WITH_HPP
+
+#include <boost/type_traits/integral_constant.hpp>
+
+#include <type_traits>
+
+namespace hpx { namespace traits
+{
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        struct is_callable_with_impl
```

Comment:
Hmmm. I think `traits::is_callable<F(std::size_t, future_type)>` should do what you expect. You can even enforce a certain return type, if needed: `traits::is_callable<F(std::size_t, future_type), void>` (this is not necessary in your case, however).